packages/@aws-cdk/custom-resource-handlers/lib/aws-stepfunctions-tasks/cross-region-aws-sdk-handler/index.ts

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable @typescript-eslint/no-require-imports */
+/* eslint-disable no-console */
+
+type Event = {
+  /**
+   * AWS Region name to call the service
+   */
+  region: string;
+
+  /**
+   * AWS service name (must be in AWS SDK for JS v3 style)
+   */
+  service: string;
+
+  /**
+   * AWS API endpoint
+   */
+  endpoint: string;
+
+  /**
+   * API Action name (must be in camelCase or PascalCase)
+   */
+  action: string;
+
+  /**
+   * request parameters to call the API (must be in AWS SDK for JS v3 style)
+   */
+  parameters?: any;
+};
+
+// reference: code from CDK AwsCustomResource
+function findV3ClientConstructor(pkg: object) {
+  const [_clientName, ServiceClient] = Object.entries(pkg).find(([name]) => {
+    // Services expose a base __Client class that we don't want ever
+    return name.endsWith('Client') && name !== '__Client';
+  }) as [
+    string,
+    {
+      new (config: any): {
+        send: (command: any) => Promise<any>;
+        config: any;
+      };
+    },
+  ];
+  return ServiceClient;
+}
+
+function findCommandClass(pkg: object, action: string) {
+  const camelToPascal = (str: string) => str.charAt(0).toUpperCase() + str.slice(1);
+  const commandName = `${camelToPascal(action)}Command`;
+  const Command = Object.entries(pkg).find(([name]) => name.toLowerCase() === commandName.toLowerCase())?.[1] as {
+    new (input: any): any;
+  };
+  if (!Command) {
+    throw new Error(`Unable to find command named: ${commandName} for action: ${action} in service package ${pkg}`);
+  }
+  return Command;
+}
+
+export const handler = async (event: Event) => {
+  // Log the event to understand the incoming request structure
+  console.log('Event: ', event);
+
+  try {
+    // esbuild-disable unsupported-require-call -- not esbuildable but that's fine
+    const pkg = require(`@aws-sdk/client-${event.service}`);
+    const Client = findV3ClientConstructor(pkg);
+    const Command = findCommandClass(pkg, event.action);
```

Comment:
Similarly, you should be able to use `normalizeActionName`, see [AwsSdkCall](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.custom_resources.AwsSdkCall.html#action):
https://github.com/aws/aws-cdk/blob/fff9cf694b14811682c8671a1e55afa53151df8b/packages/@aws-cdk/aws-custom-resource-sdk-adapter/lib/sdk-info.ts#L10-L16

Reply:
Same as service name, I'd like to intentionally only support action names in camelCase.